File: 3. Longest Substring Without Repeating Characters/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        string sub_str = "";
        int result = 0;
        for (int i = 0; i < s.length(); i++) {
            size_t idx = sub_str.find(s[i]);
            if (idx != string::npos){
                sub_str = sub_str.substr(idx+1,string::npos);
                sub_str += s[i]; 
            }else{
                sub_str += s[i]; 
            }
            if (sub_str.size() > result){
                result = sub_str.size();
            }
        }
        return result;
    }
};

File: 3. Longest Substring Without Repeating Characters/Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        sub_str = ''
        result = 0
        for i in range(len(s)):
            if s[i] in sub_str:
                idx = sub_str.index(s[i])
                sub_str = sub_str[idx+1:]
                sub_str += s[i]
                
            else: sub_str+=s[i]

            if len(sub_str) > result:
                result = len(sub_str)
        return result